temporalio/lib/temporalio/internal/worker/multi_runner.rb

```diff
@@ -0,0 +1,161 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'temporalio/internal/bridge/worker'
+
+module Temporalio
+  module Internal
+    module Worker
+      class MultiRunner
+        def initialize(workers:)
+          @workers = workers
+          @queue = Queue.new
+
+          @shutdown_initiated_mutex = Mutex.new
+          @shutdown_initiated = false
+
+          # Start pollers
+          Bridge::Worker.async_poll_all(workers.map(&:_bridge_worker), @queue)
+        end
+
+        def apply_thread_or_fiber_block(&)
+          return unless block_given?
+
+          @thread_or_fiber = if Fiber.current_scheduler
+                               Fiber.schedule do
+                                 @queue.push(Event::BlockSuccess.new(result: yield))
+                               rescue InjectEventForTesting => e
```

Comment:
1. Why do we need to push two events in this case?
2. It looks like atm this is only used to simulate a Rust-side error; is the intention here that this might be used to inject other type of events? If so, shouldn't there be a loop here?

Reply:
> Why do we need to push two events in this case?

Because the block is complete too when this is raised, we don't want to pretend the block is still running.

> It looks like atm this is only used to simulate a Rust-side error; is the intention here that this might be used to inject other type of events? If so, shouldn't there be a loop here?

Yes, that's the intention to simulate Rust side polling error (hard to simulate otherwise). Our need is just for injecting a single event then completing the block. A loop would cause the block to run a _second_ time which we don't want.